zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -43,6 +43,7 @@ use zeitgeist_primitives::{
         SubsidyUntil,
     },
 };
+use zrml_authorized::Pallet as AuthorizedPallet;
```

Comment:
> Something seems off here. Although `zrml_authorized` is not tightly coupled to `zrml_predition_markets`, we act here as if this exact implementation is. I think a cleaner approach would be to allow any authorized implementation to be used in the benchmarks.
> We could add `authorize_market_outcome` to `AuthorizedPalletApi` in `zrml_authorized` and then use the `Authorized` from prediction markets to call `authorize_market_outcome`.

Good point. I have to admit that I've been pretty nonchalant about this in benchmarks. Definitely better to have a clear interface.

> Not 100% sure how we can elegantly bypass the origin check in `authorize_market_outcome`, currently the only way I see is to add a `successful_origin` function to the `AuthorizedPalletApi` under the `[cfg(test)]` environment.

I strongly advise against using `[cfg(test)]`. But what about `[cfg(feature = runtime_benchmarks)]`? On second thought, can't we add the `AuthorizedOrigin` _type_ (or whatever it's called) to the `AuthorizedPalletApi`?

Reply:
Should this go in a separate PR? In that case, I'll open an issue.

Reply:
> I strongly advise against using `[cfg(test)]`. But what about `[cfg(feature = runtime_benchmarks)]`? On second thought, can't we add the `AuthorizedOrigin` _type_ (or whatever it's called) to the `AuthorizedPalletApi`?

I think adding an associated type to the trait, as you suggested, is a second viable variant. We would [have all necessary functions availble](https://paritytech.github.io/substrate/master/frame_support/traits/trait.OriginTrait.html) through that.

Probably a combination of both approaches is the solution we are looking for. We'd have an associated type for the origin within the authorized trait. The trait offers a function (under proper cfg environment) to return the origin using the associated origin type. This gives us flexibility to use and return whatever origin we like, while at the same time allowing us to efficiently return the origin on stack instead of the heap by using trait objects.

> Should this go in a separate PR? In that case, I'll open an issue.

Yes, please.

Reply:
https://github.com/zeitgeistpm/zeitgeist/issues/933

Reply:
Another way of looking at this is that the problem here is that the benchmarks are using the actual pallets. Using a mock and letting the pallets return the weights of their calls seems to resolve the issue.